tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java

```diff
@@ -59,21 +59,16 @@ public void setRollbackRules(Set<RollbackRule> rollbackRules) {
     }
 
     public boolean rollbackOn(Throwable ex) {
-
         RollbackRule winner = null;
-        int deepest = Integer.MAX_VALUE;
-
         if (CollectionUtils.isNotEmpty(rollbackRules)) {
             winner = NoRollbackRule.DEFAULT_NO_ROLLBACK_RULE;
             for (RollbackRule rule : this.rollbackRules) {
                 int depth = rule.getDepth(ex);
-                if (depth >= 0 && depth < deepest) {
-                    deepest = depth;
+                if (depth >= 0) {
```

Comment:
rollbackRules contains AException and BException,  AException extend BException.  please valid this scenario.

Reply:
add class MyIllegalArgumentException extends IllegalArgumentException,and test is ok

Reply:
> rollbackRules contains AException and BException, AException extend BException. please valid this scenario.

add class MyIllegalArgumentException extends IllegalArgumentException,and test is ok

Reply:
> rollbackRules contains AException and BException, AException extend BException. please valid this scenario.
add MyIllegalArgumentException extends IllegalArgumentException ,test is ok.
and I have a question below.

Reply:
It is always the smallest win. You have not compared their priority here. What is the point of the loop just assigning?

Reply:
yes, return value is boolean ,Why compare priorities？if roolback and noroolback has same priority, i think below can add break for reducing cycle times...

Reply:
sorry, just thought if roolbackrule and noroolnack has father and son exception, smallest win will useful